src/libraries/System.Net.Http/src/System/Net/Http/HttpMetrics.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpMetrics
+    {
+        private readonly Meter _meter;
+        private readonly UpDownCounter<long> _currentRequests;
+        private readonly Histogram<double> _requestsDuration;
+
+        public HttpMetrics(Meter meter)
+        {
+            _meter = meter;
+
+            _currentRequests = _meter.CreateUpDownCounter<long>(
+                "current-requests",
+                description: "Number of outbound HTTP requests that are currently active on the client.");
+
+            _requestsDuration = _meter.CreateHistogram<double>(
+                "request-duration",
+                unit: "s",
```

Comment:
Is the issue specifically that the unit was specified? For example if the code didn't specify a unit but still captured measurements in seconds would that solve the problem? My understanding is that it would not.

I think emitting the measurement in seconds (and marking the unit that way) is the right thing for .NET to do because it appears to be the overall approach industry at large is standardizing on. However I also get that not all consumers are aligned behind a single standard and in particular we have MS 1P customers who want ms measurements for compat with their existing practice. I believe we need to invest in techniques that let us present different views of the data to different customer segments. A couple ideas how that might happen:

1. Creating a new instrument with different naming and only exporting data from the new instrument.
2. Ignoring this instrument and continuing to collect a separate measurement via a different handler or DiagnosticSource
3. Doing work in OTel support changing the units on an existing instrument before it is reported, perhaps as part of a View
4. Doing a similar feature as (3) but in the runtime further building on the [config and pipeline work](https://github.com/dotnet/runtime/issues/85684)

(1) or (2) seem like they might be our shortest-term options but (3) and (4) are probably where I'd hope we end up eventually.